fxa-auth-db-server/test/backend/db_tests.js

```diff
@@ -2180,11 +2180,22 @@ module.exports = function(config, DB) {
                     .then(function () {
                       testAccount.email = testSecondaryEmail.email
                       testAccount.normalizedEmail = testSecondaryEmail.normalizedEmail
+
+                      // Create new account with secondary email that was deleted
                       return db.createAccount(testAccount.uid, testAccount)
                         .catch(t.fail)
                     })
                     .then(function (res) {
                       t.deepEqual(res, {}, 'successfully created an account')
+
+                      // Attempt to create secondary email address
+                      return db.createEmail(testAccount.uid, testSecondaryEmail)
+                        .then(() => {
+                          t.fail(new Error('Should not have created secondary email'))
```

Comment:
I think it should? The newly created account is using the email address of the secondary email. This is the flow these test cases are going through. I could be missing something.

* Create account A (a@a.com)
* Add secondary email (b@b.com) to account A
* Delete account A (a@a.com)
* Create account B (b@b.com), this was previously failing, but now works
* Create secondary email (b@b.com) on account B, fails because address belongs to account